event-handler/configure_cmd.go

```diff
@@ -438,23 +401,81 @@ func (c *ConfigureCmd) writeConf() error {
 func (c *ConfigureCmd) askOverwrite(path string) bool {
 	_, err := os.Stat(path)
 	if !os.IsNotExist(err) {
-		return c.yesNo(fmt.Sprintf("Do you want to overwrite %s", path))
+		return lib.AskYesNo(fmt.Sprintf("Do you want to overwrite %s", path))
 	}
 
 	return true
 }
 
-// yesNo displays Y/N prompt
-func (c *ConfigureCmd) yesNo(message string) bool {
-	prompt := promptui.Prompt{
-		Label:     message,
-		IsConfirm: true,
+// genCertAndPK generates and returns certificate and primary key
+func (c *ConfigureCmd) genCertAndPK(cert *x509.Certificate, parent *x509.Certificate, signer *rsa.PrivateKey) (*rsa.PrivateKey, []byte, error) {
+	sn, err := rand.Int(rand.Reader, maxBigInt)
+	if err != nil {
+		return nil, nil, trace.Wrap(err)
 	}
 
-	result, err := prompt.Run()
+	cert.SerialNumber = sn                // Assign generated serial number
+	cert.NotAfter = time.Now().Add(c.TTL) // Assign expiration time
+
+	// Generate PK
+	pk, err := rsa.GenerateKey(rand.Reader, c.Length)
 	if err != nil {
-		return false
+		return nil, nil, trace.Wrap(err)
+	}
+
+	// Check if it's self-signed, assign signer and parent to self
+	s := signer
+	p := parent
+
+	if s == nil {
+		s = pk
 	}
 
-	return result == "y"
+	if p == nil {
+		p = cert
+	}
+
+	// Generate and sign cert
+	certBytes, err := x509.CreateCertificate(rand.Reader, cert, p, &pk.PublicKey, s)
+	if err != nil {
+		return nil, nil, trace.Wrap(err)
+	}
+
+	return pk, certBytes, nil
+}
+
+// writeKeyAndCert writes private key and certificate on disk, returns file names actually written
+func (c *ConfigureCmd) writeKeyAndCert(certAndKeyPaths []string, certBytes []byte, pk *rsa.PrivateKey, pwd string) error {
+	var err error
+
+	ok := c.askOverwrite(certAndKeyPaths[0])
```

Comment:
Fair enough, fixed.

Reply:
It's still in genCerts, which is one level done. I think, it would not be a good idea to move this conditions to Run() itself because it might make Run() just too long.